cloudinit/cmd/main.py

```diff
@@ -461,133 +462,150 @@ def main_init(name, args):
-            if mode == sources.DSMODE_LOCAL:
+            LOG.debug("Exiting without datasource")
+            if stage == stages.local:
                 return (None, [])
             else:
                 return (None, ["No instance datasource found."])
         else:
-            LOG.debug(
-                "[%s] barreling on in force mode without datasource", mode
-            )
+            # XXX: TODO
+            #
+            # --force is untested and broken. It assumes that something
+            # useful can be accomplished without a datasource object.
+            #
+            # The next use of init.datasource or init.fetch(), such as in
+            # _maybe_persist_instance_data(), will throw an exception. Consider
+            # removing --force, or perhaps to avoid upsetting users that
+            # currently use it, just change the code path to be the same with
+            # or without --force.
+            LOG.debug("Barreling on in force mode without datasource.")
 
     _maybe_persist_instance_data(init)
     # Stage 6
     iid = init.instancify()
     LOG.debug(
-        "[%s] %s will now be targeting instance id: %s. new=%s",
-        mode,
-        name,
+        "%s will now be targeting instance id: %s. new=%s",
+        stage.short,
         iid,
         init.is_new_instance(),
     )
 
-    if mode == sources.DSMODE_LOCAL:
+    if stage == stages.local:
         # Before network comes up, set any configured hostname to allow
         # dhcp clients to advertize this hostname to any DDNS services
         # LP: #1746455.
-        _maybe_set_hostname(init, stage="local", retry_stage="network")
-
-    init.apply_network_config(bring_up=bring_up_interfaces)
-
-    if mode == sources.DSMODE_LOCAL:
+        _maybe_set_hostname(
+            init,
+            stage=stages.local.internal,
+            retry_stage=stages.network.internal,
+        )
+        init.apply_network_config(bring_up=bring_up_interfaces)
         should_wait, reason = _should_wait_on_network(init.datasource)
         if should_wait:
             LOG.debug(
                 "Network connectivity determined necessary for "
-                "cloud-init's network stage. Reason: %s",
+                "cloud-init's %s. Reason: %s",
+                stages.network.long,
                 reason,
             )
         else:
             LOG.debug(
                 "Network connectivity determined unnecessary for "
-                "cloud-init's network stage. Reason: %s",
+                "cloud-init's %s. Reason: %s",
+                stages.network.long,
                 reason,
             )
             util.write_file(init.paths.get_runpath(".skip-network"), "")
 
-        if init.datasource.dsmode != mode:
+        # see comment above regarding dsmode
+        if init.datasource.dsmode != sources.DSMODE_LOCAL:
             LOG.debug(
-                "[%s] Exiting. datasource %s not in local mode.",
```

Comment:
The `[%s]` was redundant. If the datasource is not in local mode, then it is in network mode. The surrounding logs also hold this information.